_schema.graphql

```diff
@@ -77,7 +77,11 @@ type CollectionQuery {
 scalar DateTime
 
 type Query {
-  collections(showOnEditorial: Boolean, artistID: String): [Collection!]!
+  collections(
+    size: Float
```

Comment:
As commented on the related MP PR, should this be an integer?

Reply:
Probably, but when the schema was generated here it created `Float`. That comes from here: https://github.com/artsy/kaws/pull/62/files#diff-500113cce4e1c0ce1050b25c61eb35a4R14

`size: number`

Was not able to figure out with this library how to change that? Since it gets derived from the TS typings (where there is no integer, right?).

Overall this library was a bit of a challenge to use and figure out. I get what you're saying...but I _think_ this is fine (plus couldn't figure out anything else...so would love another pair of eyes if you're volunteering!). Otherwise we'll roll with this so as not to block on a type thing. We can always change it later.

Reply:
Yeah, looks like the library must prefer float representation - it does that here also. https://github.com/artsy/kaws/blob/master/_schema.graphql#L31-L32

Reply:
Well, that’s not great… searching [the repo](https://github.com/19majkel94/type-graphql) I can’t even find a single mention about this.

I'm not of the opinion that this PR should be blocked, but I do think that long term the schema should be self-documenting and not be confusing due to some JS implementation details (correct, there is no integer JS type). The change from `Float` to `Int` will technically be a breaking one, but as long as the app isn't using this it should probably be fine (although it will probably require a concerted effort).

@l2succes Did you run into this yet and consider it?

Reply:
I ran into a similar issue on `price_guidance`, which resulted in a bug when added to Metaphysics -- https://artsyproduct.atlassian.net/browse/GROW-1126

This field is supposed to be an optional `number`, but the schema labels it as a `Float`.  When empty, it results in the error that a float cannot be empty.  Not sure if this is directly related, but an unexpected result of the typecasting from this library.

Reply:
Yes, you can explicitly specify the type of a graphQL argument. The following way
```typescript
import { Arg, Int, Query, Resolver } from "type-graphql"

@Query(returns => [Collection])
  async collections(
    ...
    @Arg("size", () => Int, { nullable: true }) size: number
  ): Promise<Collection[]> {
  ...
}
```


This will generate the following SDL.
```graphql
type Query {
  collections(
    size: Int
  ): [Collection!]!
}
```

`type-graphql` actually has really good documentation, you can read more about how to set types here
https://19majkel94.github.io/type-graphql/docs/resolvers.html#arguments

Let me know if you have any more questions

Reply:
Thanks!